x/finality/keeper/power_dist_change.go

```diff
@@ -264,24 +283,29 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 			fp.IsJailed = false
 		}
 
-		// add all BTC delegations that are not unbonded to the new finality provider
-		for j := range dc.FinalityProviders[i].BtcDels {
-			btcDel := *dc.FinalityProviders[i].BtcDels[j]
-			if _, ok := unbondedBTCDels[btcDel.StakingTxHash]; !ok {
-				fp.AddBTCDelDistInfo(&btcDel)
+		// process all new BTC delegations under this finality provider
+		if fpUnbondedBTCDels, ok := unbondedSatsByFpBtcPk[fpBTCPKHex]; ok {
+			// handle unbonded delegations for this finality provider
+			for _, unbodedSats := range fpUnbondedBTCDels {
+				fp.RemoveBondedSats(unbodedSats)
```

Comment:
here we process unbondedSats earlier than bondedSats, will it make the sats overflow (uin64 cannot be a negative value)? Better have a test case for it

Reply:
modified to handle active first, test should be added either way in another PR :sweat_smile: